libdevice/sanitizer_utils.cpp

```diff
@@ -281,35 +301,36 @@ bool MemIsZero(__SYCL_GLOBAL__ const char *beg, uptr size) {
 /// ASAN Save Report
 ///
 
-bool __asan_internal_report_save(DeviceSanitizerErrorType error_type) {
+bool __asan_internal_report_save(DeviceSanitizerErrorType error_type,
+                                 uptr launch_info) {
   const int Expected = ASAN_REPORT_NONE;
   int Desired = ASAN_REPORT_START;
-  if (atomicCompareAndSet(&__DeviceSanitizerReportMem.get().Flag, Desired,
-                          Expected) == Expected) {
-    __DeviceSanitizerReportMem.get().ErrorType = error_type;
+  auto *SanitizerReport =
+      &((__SYCL_GLOBAL__ LaunchInfo *)launch_info)->SanitizerReport;
+  if (atomicCompareAndSet(&SanitizerReport->Flag, Desired, Expected) ==
+      Expected) {
+    SanitizerReport->ErrorType = error_type;
     // Show we've done copying
-    atomicStore(&__DeviceSanitizerReportMem.get().Flag, ASAN_REPORT_FINISH);
+    atomicStore(&SanitizerReport->Flag, ASAN_REPORT_FINISH);
     return true;
   }
   return false;
 }
 
-#ifdef __SYCL_DEVICE_ONLY__
-#define __DEVICE_SANITIZER_REPORT_ACCESSOR __DeviceSanitizerReportMem.get()
-#else
-#define __DEVICE_SANITIZER_REPORT_ACCESSOR
-#endif
-
 bool __asan_internal_report_save(
-    uptr ptr, uint32_t as, const char __SYCL_CONSTANT__ *file, uint32_t line,
-    const char __SYCL_CONSTANT__ *func, bool is_write, uint32_t access_size,
-    DeviceSanitizerMemoryType memory_type, DeviceSanitizerErrorType error_type,
-    bool is_recover = false) {
+    uptr ptr, uint32_t as, uptr launch_info, const char __SYCL_CONSTANT__ *file,
+    uint32_t line, const char __SYCL_CONSTANT__ *func, bool is_write,
+    uint32_t access_size, DeviceSanitizerMemoryType memory_type,
+    DeviceSanitizerErrorType error_type, bool is_recover = false) {
 
   const int Expected = ASAN_REPORT_NONE;
   int Desired = ASAN_REPORT_START;
-  if (atomicCompareAndSet(&__DEVICE_SANITIZER_REPORT_ACCESSOR.Flag, Desired,
-                          Expected) == Expected) {
+
+  auto *plaunch = (__SYCL_GLOBAL__ LaunchInfo *)launch_info;
```

Comment:
`plaunch` -> `pLaunch` maybe?

Reply:
this is too minor... I used "plaunch" throughout this code.

Reply:
"plaunch" removed